lib/modern_searchlogic/column_conditions.rb

```diff
@@ -84,8 +92,17 @@ def searchlogic_suffix_condition_match(method_name)
             arity: arity,
             block: lambda do |*args|
               validate_argument_count!(arity, args.length) if arity >= 0
-              arel_conditions = column_names.map { |n| instance_exec(n, *args, &method_block) }.reduce(:or)
-              where(arel_conditions)
+              partial_queries = column_names.map do |n|
+                query = unscoped { instance_exec(n, *args, &method_block) }
+                if ActiveRecord::VERSION::MAJOR >= 5 && query.respond_to?(:where_clause)
```

Comment:
let's move all of the code that has to worry about active record relations into the `searchlogic_active_record_alias` method, so that we're consistently passing arel nodes back to `searchlogic_suffix_condition_match` and we can contain that complexity

Reply:
that also lets us get rid of the respond to and the complicated "either an active record relation OR an arel node" logic